File: LeetCode/twoSum.py
# 1

# Given an array of integers, return indices of the two numbers such that they add up to a specific target.
#
# You may assume that each input would have exactly one solution, and you may not use the same element twice.

def twoSum(nums, target):
    """
    :type nums: List[int]
    :type target: int
    :rtype: List[int]
    """

    seen = {}
    output = []

    for idx, number in enumerate(nums):
        goal = target - number
        if goal not in seen:
            seen[number] = idx
        else:
            output.append(seen[goal])
            output.append(idx)
    return output

File: LeetCode/CSharp/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) 
    {
        var dict = new Dictionary<int, int>();

        for(int i = 0; i < nums.Length; i++)
        {
            int lookFor = target - nums[i];
            if(dict.ContainsKey(lookFor)) return new int[] {dict[lookFor], i};
            else dict[nums[i]] = i;
        }

        return new int[]{0,0};
    }

    public static void Main(string[] args) 
    {
        var nums = new int[] {1,2,3,4};
        
        int target = 5;
        
        var answer = TwoSum(nums, target);
        foreach(var item in answer)
        {
            Console.WriteLine(item);
        }
    }
}